src/test/resources/annoingenerictarget/expected/org/checkerframework/checker/initialization/qual/Initialized.java

```diff
@@ -0,0 +1,5 @@
+package org.checkerframework.checker.initialization.qual;
+
+@java.lang.annotation.Target({ java.lang.annotation.ElementType.TYPE_USE, java.lang.annotation.ElementType.METHOD })
```

Comment:
Why `TYPE_USE` and `METHOD`, specifically? It seems like you've hardcoded these two `ElementType`s, but annotations can be present in other places, too. Can you add a test with an annotation in a different position (e.g., on a class declaration)?

One solution to this problem is to make every synthetic annotation legal in every context. That's what I'd suggest, at least initially, as a conservative but over-approximate solution. However, it might be possible to be more precise by recording the contexts in which the annotation is actually used and then only including those; I don't think that needs to be in this PR, but we can add it as a TODO for the future.

Reply:
In testing, these two `ElementType` were sufficient for most (if not all) cases. This is probably due to this ([ElementType docs](https://docs.oracle.com/javase/8/docs/api/java/lang/annotation/ElementType.html)):

>The constant [TYPE_USE](https://docs.oracle.com/javase/8/docs/api/java/lang/annotation/ElementType.html#TYPE_USE) corresponds to the 15 type contexts in JLS 4.11, as well as to two declaration contexts: type declarations (including annotation type declarations) and type parameter declarations.

With the exception of `void` methods (hence the inclusion of `METHOD`), the annotations should be interpreted by the compiler to be applied to the type of the declaration instead of the declaration itself, which shouldn't cause any errors since there aren't such things as void field or class declarations, for example.

I've added a new test case `SyntheticAnnotationTarget` in which annotations are applied in many different contexts (like class declarations) to test for compile errors.

If you think this behavior is fine, or if I am missing something, please let me know! I can definitely work on including all `ElementType`s if necessary.

Reply:
I'm satisfied with the new test, thanks!

My only (mild) remaining concern with respect to this is that I know that annotation processors can tell whether an annotation is a type annotation or a declaration annotation, so which of the two it is can impact their behavior. For the case of unsolved annotations, however, there's no way to resolve this ambiguity so I think the approach you're taking is fine.